backend/flask_api.py

```diff
@@ -73,17 +73,19 @@ def add_project():
             if request.json:
                 return_ids = uploader.save_manifest_to_db(request.json)
 
-            else:
+            elif request.data:
                 print(request.data.decode("utf-8"), file=sys.stderr)
                 return_ids = uploader.save_manifest_to_db(
                     json5.loads(request.data.decode("utf-8")))
-                print(return_ids)
+
+            else:
+                return make_response("Error: empty POST body", 400)
 
             return jsonify(return_ids)
         except ApiException as e:
             raise e
         except Exception as err:
-            return make_response("error: " + str(err), '500')
+            return make_response("Error: " + str(err), 500)
```

Comment:
This should not be necessary. An uncatched exception always results in a 500 with the appropriate message, please remove.